distributed/security.py

```diff
@@ -123,7 +123,8 @@ def get_connection_args(self, role):
         """
         d = {}
         tls = self.get_tls_config_for_role(role)
-        d['ssl_context'] = self._get_tls_context(tls, ssl.Purpose.SERVER_AUTH)
+        purpose = ssl.Purpose.SERVER_AUTH if hasattr(ssl, "Purpose") else None
+        d['ssl_context'] = self._get_tls_context(tls, purpose)
```

Comment:
I had a quick check: All other usages (there are only 3 if I didn't miss anything) should be okay and covered. How do you prefer to communicate the error? I went for raising a runtime error with a more explicit message.